src/payment/strategies/humm/humm-payment-strategy.ts

```diff
@@ -0,0 +1,54 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { some } from 'lodash';
+
+import { CheckoutStore,  InternalCheckoutSelectors } from '../../../checkout';
+import { RequestError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+export default class HummPaymentStrategy implements PaymentStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _formPoster: FormPoster
+    ) {}
+
+    async initialize(): Promise<InternalCheckoutSelectors> {
+        return this._store.getState();
+    }
+
+    async deinitialize(): Promise<InternalCheckoutSelectors> {
+        return this._store.getState();
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        if (!payment?.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        try {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId: payment.methodId}));
+        } catch (error) {
+            return new Promise(() => {
+                if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'}) || !error.body.provider_data) {
```

Comment:
the `ExternalStrategy` already has a method to validate if the error is because of an additional action: `_isAdditionalActionRequired`